### ISSUE TITLE
icns not visible in macOS Sonoma's We-are-switching-keyboards badge attached to text fields when that this layout is selected

### ISSUE THREAD

**Issue by thomashankeuhh** [felixfoertsch/EurKEY-macOS#10]
Congrats to making the icon look great in light and dark mode. However, with Sonoma & Sequoia there is a new challenge: The badge the system attaches to text fields when cmd-spacing thru available layouts. Here the stars show as long as they are not selected. I have no idea if this is a bug in macOS. The switcher works with the predefined keyboard layouts as well as non-template icons, but maybe not with template icons?
<img width="224" alt=" selected" src="https://github.com/user-attachments/assets/733f1a35-2d0f-4f1f-bf71-51d60e0733c6">
<img width="274" alt="notselected" src="https://github.com/user-attachments/assets/63ab1129-c710-482e-a8cf-f4578293beba">

**Comment by felixfoertsch** [comment]
Hello @thomashankeuhh! I am very sorry I just saw that you opened an issue here today. I will check that GitHub sends me emails, so I actually notice them.

I don't think I've ever seen the bar you are talking about. How do I bring it up? If I have time, I might see what I can do. :)

**Comment by thomashankeuhh** [comment]
In Sonoma or Sequoia, put the cursor into Safari's search field and press cmd-space. Instead of seeing the relatively large light-gray window centered on screen, you see the badge as shown in my first post.

**Comment by felixfoertsch** [comment]
@thomashankeuhh I looked into it a little. It's unclear to me why the template icon does not work in this specific context. All my devices are on the latest Sequoia Developer Beta. From what you've wrote, you experienced this issue on Sonoma as well? Did I understand you correctly?

**Comment by thomashankeuhh** [comment]
Exactly: I did not see any difference between Sonoma (which introduced these badges) and Sequoia. And my sympathy for your statement “It's unclear to me”. To me as well, and I am inclined to think that this is a bug in macOS.

**Comment by felixfoertsch** [comment]
This is a macOS bug affecting third-party keyboard layouts that use template icons (`TISIconIsTemplate = true`). Template icons correctly adapt to light/dark mode in the menu bar, but disappear in the input source switching badge attached to text fields (introduced in Sonoma).

Apple's built-in layouts are not affected. Non-template icons work correctly in the badge but lose dark mode adaptation.

Documented as a known issue in the README. Closing as upstream/wontfix — this requires a fix from Apple.
